src/main/lock.ts

```diff
@@ -0,0 +1,33 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { app } from "electron";
+import { container } from "readium-desktop/main/di";
+import { WinRegistry } from "readium-desktop/main/services/win-registry";
+
+export function lockInstance() {
+    const gotTheLock = app.requestSingleInstanceLock();
+
+    if (!gotTheLock) {
+        app.quit();
+        return true;
+    } else {
+        app.on("second-instance", () => {
```

Comment:
The `argv` handling is missing.
See my full example in the issue tracker to visualize the darwin / linux+windows flow of actions:
https://github.com/readium/readium-desktop/issues/547

Reply:
I thinks it's done, test on your side